### ISSUE TITLE
beancount fails with ValueError: day is out of range for month on document

### ISSUE THREAD

**Issue by blais** [beancount/beancount#141]
**[Original report](https://bitbucket.org/blais/beancount/issue/141) by Renato Alves.**

----------------------------------------

This was a tricky one.

The error:

```
#!python
  ...
  File "/virtualenvs/beancount/lib/python3.4/site-packages/beancount/ops/documents.py", line 141, in find_documents
    date = datetime.date(*map(int, match.group(1, 2, 3)))
ValueError: day is out of range for month
```

Initially I thought I had mistyped a date in my document. Turned out the cause was a document named `2016-09-31.doc.pdf`.

Obviously beancount is correct in that the month 9 only has 30 days, therefore 31 is invalid. However it crashes with a traceback that provides no helpful information of what is wrong. Be it a line in the document or in this case, the name of the offending file.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Thanks for reporting this Renato, I'll address it.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Fixed #141: Invalid date on document should be caught and produce a nice error message.

**Comment by blais** [comment]
**Original comment by Renato Alves.**

----------------------------------------

Thanks for the quick fix!
